### PR TITLE
1.14/1847 wip2

### PR DESCRIPTION
**Pull request by charlesstrange2**
quite a substantial rewrite

covers 
https://trello.com/c/H0vSMbIR/1847-we-have-components-for-age-and-or-monthyear-vv-ee
https://trello.com/c/23TMeN57/1813-taking-age-of-all-other-household-members-vvv-ee

### CODE REVIEW

**Review by neonleann — CHANGES_REQUESTED** [review]
Things that aren't working as expected:
Invalid date messages don't disappear when a valid age is entered.
List of carers isn't saving (Laura tells me it does work when editing, just not creating).
You can put any old nonsense as a carer name.
You can add someone who is 999yrs old.
If you put a really high number in, it tells you you have a P.
Apart from the last one, which we dind't check on Chrome, all the rest occur in FF and Chrome.

File: resources/views/store/partials/householdSP.blade.php [hunk 1]
```diff
@@ -6,27 +10,102 @@
...
+            var valueDate = dateObj.format("YYYY-MM");
+
+            // give the kids a random key
+            var childKey = Math.random();
+
+            // Create the table columns
+            var ageColumn = (moment().diff(valueDate, 'days') > 0)
+                ? '<td class="age-col">' +
+                moment().diff(valueDate, 'years') + ' yr, ' +
+                moment().diff(valueDate, 'months') % 12 +
```
Comment (charlesstrange2) [inline resources/views/store/partials/householdSP.blade.php:73]: this looks wierd, remove months

**Review by lauragurney — APPROVED** [review]
Yes 🐈‍⬛

**Review by neonleann — APPROVED** [review]
Approved conditionally so we can check it once it's merged.

**Pull request merged**